tsl/test/sql/dist_hypertable.sql.in

```diff
@@ -359,18 +359,60 @@ SELECT * FROM disttable;
 
 -- Test INSERTS with RETURNING. Since we previously dropped a column
 -- on the hypertable, this also tests that we handle conversion of the
--- attribute numbers in the RETURNING clause, since they now differ
+-- attribute numbers in the RETURNING clause since they now differ
 -- between the hypertable root relation and the chunk currently
--- RETURNING from.
+-- RETURNING from. Should use a DataNodeCopy plan when no triggers
+-- exists that can potentially modify tuples on INSERT.
+EXPLAIN VERBOSE
 INSERT INTO disttable (time, device, "Color", temp)
 VALUES ('2017-09-02 06:09', 4, 1, 9.8)
 RETURNING time, "Color", temp;
 
+-- Dropping the triggers should allow RETURNING with DataNodeCopy
+DROP TRIGGER _0_test_trigger_insert ON disttable;
+DROP TRIGGER _1_test_trigger_insert ON disttable;
+
+EXPLAIN VERBOSE
+INSERT INTO disttable (time, device, "Color", temp)
+VALUES ('2017-09-02 06:09', 4, 1, 9.8)
+RETURNING time, "Color", temp;
+
+-- Check that it is possible to disable DataNodeCopy with a session
+-- variable.
+SET timescaledb.enable_distributed_insert_with_copy=false;
+
+EXPLAIN VERBOSE
+INSERT INTO disttable (time, device, "Color", temp)
+VALUES ('2017-09-02 06:09', 4, 1, 9.8)
+RETURNING time, "Color", temp;
+
+-- Re-enable and do the actual insert.
+SET timescaledb.enable_distributed_insert_with_copy=true;
+
+INSERT INTO disttable (time, device, "Color", temp)
+VALUES ('2017-09-02 06:09', 4, 1, 9.8)
+RETURNING time, "Color", temp;
+
+-- Recreate the triggers
+CREATE TRIGGER _0_test_trigger_insert
+    BEFORE INSERT ON disttable
+    FOR EACH ROW EXECUTE FUNCTION test_trigger();
+
+CREATE TRIGGER _1_test_trigger_insert
+    BEFORE INSERT ON disttable
+    FOR EACH ROW EXECUTE FUNCTION test_trigger();
+
 INSERT INTO disttable (time, device, "Color", temp)
 VALUES ('2017-09-03 06:18', 9, 3, 8.7)
 RETURNING 1;
 
--- On conflict
+-- On conflict. Should use a DataNodeDispatch plan and NOT a
```

Comment:
I believe there should be no impact on using COPY to insert into compressed chunks? Do we need a test case to cover inserting into a compressed chunk?

Reply:
Sven pointed out that COPY to compressed chunks does not work and I believe it's been tracked elsewhere. So resolving this here for now.